packages/@aws-cdk/toolkit-lib/lib/toolkit/toolkit.ts

```diff
@@ -268,122 +266,40 @@ export class Toolkit extends CloudAssemblySourceBuilder implements AsyncDisposab
 
     const strict = !!options.strict;
     const contextLines = options.contextLines || 3;
-    const diffMethod = options.method ?? DiffMethod.ChangeSet();
 
     let diffs = 0;
     let formattedSecurityDiff = '';
     let formattedStackDiff = '';
 
-    if (diffMethod.method === 'local-file') {
-      const methodOptions = diffMethod.options as LocalFileDiffOptions;
+    const templateInfos = await makeTemplateInfos(ioHelper, stacks, deployments, await this.sdkProvider('diff'), options);
```

Comment:
So this is preparing the diff? Current names read unclear to me. Maybe something like this is clearer:

```suggestion
    const diffInput = await prepareDiff(ioHelper, stacks, deployments, await this.sdkProvider('diff'), options);
```

Reply:
i like the prepareDiff name but also after my latest refactor i think the type of `TemplateInfo` now makes more sense -- since its only getting passed into `DiffFormatter`. lemme know if you disagree